plugins/database/mysql/connection_producer_test.go

```diff
@@ -0,0 +1,295 @@
+package mysql
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"io/ioutil"
+	"os"
+	paths "path"
+	"path/filepath"
+	"reflect"
+	"testing"
+	"time"
+
+	"github.com/hashicorp/vault/helper/testhelpers/certhelpers"
+	"github.com/hashicorp/vault/sdk/database/helper/dbutil"
+	"github.com/ory/dockertest"
+)
+
+func Test_finalizeConnectionURL(t *testing.T) {
+	type testCase struct {
+		rootUrl        string
+		tlsConfigName  string
+		expectedResult string
+	}
+
+	tests := map[string]testCase{
+		"no tls, no query string": {"user:password@tcp(localhost:3306)/test", "", "user:password@tcp(localhost:3306)/test"},
```

Comment:
Single line structs are usually harder to read so I try to avoid them except in very specific circumstances.
Improved readability:
```suggestion
		"no tls, no query string": {
			rootUrl: "user:password@tcp(localhost:3306)/test",
			tlsConfigName: "",
			expectedResult: "user:password@tcp(localhost:3306)/test",
		},
```

Reply:
Made abundantly clear.